babel/src/test/java/org/apache/calcite/test/BabelParserTest.java

```diff
@@ -692,7 +692,7 @@ private void checkParseInfixCast(String sqlType) {
     final String sql = "create table foo (bar int uppercase null casespecific, "
         + "baz varchar(30) casespecific uppercase null)";
     final String expected = "CREATE TABLE `FOO` (`BAR` INTEGER UPPERCASE "
-        + "CASESPECIFIC, `BAZ` VARCHAR(30) UPPERCASE CASESPECIFIC)";
+        + "CASESPECIFIC, `BAZ` VARCHAR(30) CASESPECIFIC UPPERCASE)";
```

Comment:
Curious - how come the casespecific and uppercase parts are switched in output here, but not in the next test? Perhaps it would be possible to modify the way ColumnAttributes are added to the list of ColumnAttributes in order to ensure the output order matches the input order?

Reply:
Before this change the unparsing had a fixed order as the unparsing logic was simply checking if booleans were checked in a fixed order. Now the attributes are added to the list as they are parsed so the order is no longer fixed. I agree that it would be nice to have a fixed order but I don't think that it's worth the increased complexity. Also I'm pretty sure that the input order does match the output order, right?

Reply:
I mean the order matches with the exception of NULL as that is treated differently.

Reply:
Oh, you are right, there are two separate declarations here with different orders, so I thought the order switched.